cartography/data/indexes.cypher

```diff
@@ -305,8 +305,7 @@ CREATE INDEX IF NOT EXISTS FOR (n:SpotlightVulnerability) ON (n.host_info_local_
 CREATE INDEX IF NOT EXISTS FOR (n:SpotlightVulnerability) ON (n.lastupdated);
 CREATE INDEX IF NOT EXISTS FOR (n:SQSQueue) ON (n.id);
 CREATE INDEX IF NOT EXISTS FOR (n:SQSQueue) ON (n.lastupdated);
-CREATE INDEX IF NOT EXISTS FOR (n:User) ON (n.arn);
-CREATE INDEX IF NOT EXISTS FOR (n:User) ON (n.lastupdated);
+CREATE INDEX IF NOT EXISTS FOR (n:UserAccount) ON (n.id);
```

Comment:
Since SSOUserSchema uses the data model, you don't need to manually update indexes for this. The index will be automatically ensured for you: https://github.com/cartography-cncf/cartography/blob/bf8d6670a9687bfb5dc40776af37b8a30cf2c910/cartography/graph/querybuilder.py#L424.

Under the hood, a function called `ensure_indexes()` runs everytime you call `load()` on a schema object: https://github.com/cartography-cncf/cartography/blob/bf8d6670a9687bfb5dc40776af37b8a30cf2c910/cartography/client/core/tx.py#L252

Reply:
I need this because I added an abstract label of UserAccount and applied to all OktaUsers. SSOUsers can map to many IDP providers, so I did not want to make AWSSSOUsers dependent on a particular IDP provider. Since Okta and other IDP provider users are not using the new load method, I need to add the index here.

Reply:
Right, and that case is taken care of because `SSOUserSchema` object has `UserAccount` in `extra_node_labels`.

So when you do 

```python
    load(
        neo4j_session,
        SSOUserSchema(),
        users,
     ...
    )
```

`load()` will call `ensure_indexes()` under the hood, and then `ensure_indexes()` will call `build_create_index_queries()`: 

https://github.com/cartography-cncf/cartography/blob/bf8d6670a9687bfb5dc40776af37b8a30cf2c910/cartography/graph/querybuilder.py#L424-L430

This will make sure that we run `CREATE INDEX IF NOT EXISTS` for every node label + `id` field. It also looks at all cases where TargetNodeMatchers are trying to match on other nodes and other fields and then makes sure an index exists for those.

So, all indexes are automatically taken care of. The indexes.cypher file is only for cases where we haven't had time to refactor older code.

Reply:
You are correct that AWSSSOUser is also :UserAccount, but that is unimportant. I need :OktaUser to also be a :UserAccount. Since OktaUser is created the old way I added the index the old way. If you don't want the index to be updated the old way I can just wait for AWSSSOUser sync to happen but then I need to wait for the index to rebuild when it should have been built during OktaUser sync.

Reply:
Gotcha, ok, that works then